### PR TITLE
db: Use tags instead of magic to hide internal jobs

### PR DESCRIPTION
**Pull request by V-FEXrt**
Manually and directly tag jobs as hidden then filter those jobs out from inspection instead of relying on magic character in the jobs commandline/label

### CODE REVIEW

**Review by ag-eitilt — CHANGES_REQUESTED** [review]
Much better than the old hack!  I do think this as it is will probably start showing `<write>`, `<mkdir>`, `<source>`, and `<claim>` jobs (caught by the previous line 160, but not decorated with the new tag yet).  I'd argue that showing `<write>` and `<mkdir>` jobs is a good thing so I'm fine with those two (though open to discussion if either of you feel otherwise), but we should probably still hide `<source>` and `<claim>`.

**Comment by V-FEXrt** [comment]
Went ahead and hid all those jobs along with the http jobs (I still consider it an implementation detail that http requests are jobs)

We can discuss `<write>`/`<mkdir>` here or separately. I don't have a strong opinion in the matter but the fact that they are mostly noise and somewhat trivially inspectable leads me to lean towards leaving them out.

aside: we could/should probably add an `--internal` flag that only matches internal jobs

File: share/wake/lib/system/job.wake [hunk 1]
```diff
@@ -156,6 +156,11 @@ export def setJobTag (key: String) (value: String) (job: Job): Job =
 
     job
 
+# Set a special tag to tell db inspection that this job shouldn't be shown by default
+export def setJobInspectVisibilityHidden (job: Job): Job =
+    job
+    | setJobTag "inspect.visibility" "hidden"
```
Comment (ag-eitilt) [inline share/wake/lib/system/job.wake:162]: Good call.

File: share/wake/lib/system/io.wake
Comment (ag-eitilt) [inline share/wake/lib/system/io.wake:1]: I'd like to talk about these two in the meeting before merging the PR.

**Review by ag-eitilt — APPROVED** [review]

File: share/wake/lib/system/io.wake
Comment (ag-eitilt) [reply share/wake/lib/system/io.wake:1]: Demonstrated in the meeting: The filter is only applied to `--last`/`--last-used` and `--last-executed`, so any other DB queries will still display the hidden jobs; this fine-grained control was my ideal solution (but I wasn't sure it was possible), so I'm quite happy with the current state.

**Review by colinschmidt — APPROVED** [review]
LGTM

**Pull request merged**